File: 0071-simplify-path/0071-simplify-path.cpp
#include <string>
#include <vector>
#include <sstream>
#include <string_view>

class Solution {
public:
    std::string simplifyPath(std::string path) {
        std::vector<std::string> stack;
        std::stringstream ss(path);
        std::string part;
        
        // Use stringstream to split by '/'
        while (std::getline(ss, part, '/')) {
            if (part == ".." ) {
                if (!stack.empty()) stack.pop_back();
            } else if (!part.empty() && part != ".") {
                stack.push_back(part);
            }
        }
        
        // Reconstruct path
        std::string result = "";
        for (const std::string& s : stack) {
            result += "/" + s;
        }
        
        return result.empty() ? "/" : result;
    }
};

File: 0071-simplify-path/0071-simplify-path.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        # Use a list as a stack; it is more memory efficient than creating 
        # a new list from split() if the path is extremely long.
        stack = []
        for part in path.split('/'):
            if part == '..':
                if stack: stack.pop()
            elif part and part != '.':
                stack.append(part)
        return "/" + "/".join(stack)